cairo/ethereum/cancun/vm/instructions/system.cairo

```diff
@@ -125,3 +164,185 @@ func return_{range_check_ptr, evm: Evm}() -> EthereumException* {
     let ok = cast(0, EthereumException*);
     return ok;
 }
+
+func generic_create{
+    process_create_message_label: felt*,
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    evm: Evm,
+}(
+    endowment: U256,
+    contract_address: Address,
+    memory_start_position: U256,
+    memory_size: U256,
+    init_code_gas: Uint,
+) -> EthereumException* {
+    alloc_locals;
+
+    let fp_and_pc = get_fp_and_pc();
+    local __fp__: felt* = fp_and_pc.fp_val;
+
+    let memory = evm.value.memory;
+    let call_data = memory_read_bytes{memory=memory}(memory_start_position, memory_size);
+    EvmImpl.set_memory(memory);
+
+    let is_gt_two_max_code_size = is_le(2 * MAX_CODE_SIZE + 1, call_data.value.len);
+    if (is_gt_two_max_code_size != 0) {
+        tempvar err = new EthereumException(OutOfGasError);
+        return err;
+    }
+
+    let accessed_addresses = evm.value.accessed_addresses;
+    let accessed_addresses_end = cast(accessed_addresses.value.dict_ptr, DictAccess*);
+    hashdict_write{dict_ptr=accessed_addresses_end}(1, &contract_address, 1);
+
+    tempvar new_accessed_addresses = SetAddress(
+        new SetAddressStruct(
+            accessed_addresses.value.dict_ptr_start,
+            cast(accessed_addresses_end, SetAddressDictAccess*),
+        ),
+    );
+    EvmImpl.set_accessed_addresses(new_accessed_addresses);
+
+    let create_message_gas = max_message_call_gas(evm.value.gas_left);
+    let new_gas_left = evm.value.gas_left.value - create_message_gas.value;
+    EvmImpl.set_gas_left(Uint(new_gas_left));
+
+    if (evm.value.message.value.is_static.value != 0) {
+        tempvar err = new EthereumException(WriteInStaticContext);
+        return err;
+    }
+    let (empty_data: felt*) = alloc();
+    tempvar empty_data_bytes = Bytes(new BytesStruct(empty_data, 0));
+    EvmImpl.set_return_data(empty_data_bytes);
+
+    let env = evm.value.env;
+    let state = env.value.state;
+    let sender_address = evm.value.message.value.current_target;
+    let sender = get_account{state=state}(sender_address);
+
+    let (sender_balance_not_enough) = uint256_lt([sender.value.balance.value], [endowment.value]);
+    let sender_nonce_max = is_zero(sender.value.nonce.value - (2 ** 64 - 1));
```

Comment:
should we add a MAX_NONCE in constants.cairo?

Reply:
I don't believe so